expression/builtin_math.go

```diff
@@ -1975,6 +1978,10 @@ func (b *builtinTruncateIntSig) evalInt(row chunk.Row) (int64, bool, error) {
 	if d >= 0 {
 		return x, false, nil
 	}
+	// math.Pow10(d) == INF if d > 308
+	if d < -308 {
```

Comment:
Is it better using [`math.IsInf`](https://golang.org/pkg/math/#IsInf) for this judgement?

Reply:
I think not, this `if` is used for avoid the situation `-(-9223372036854775808)=-9223372036854775808`(64bit os) or 
`-(-2147483648)=-2147483648`(32bit os).  and `math.Pow10(-d)==INF` if `d < -308`,so, we can return `0` directly

Reply:
`math.IsInf` can be used to judge whether `math.Pow10(-d) is `INF`, see

```go
shiftFloat := math.Pow10(int(-d))
if math.IsInf(shiftFloat) {
    return 0, false, nil
}
shift := int64(shift)
```

Reply:
the problem is not about `INF`, `math.Pow10(int(-d))=0` if `d=int.MIN_VALUE`, it cause `divide by zero` error. since `int.MIN_VALUE` is different in different os (64bit or 32bit), I make a judgement `d<-308`. we can remove this `if` statement if go can do this
```go
if d==int.MIN_VALUE {
    return 0, false, nil
}
```

Reply:
Thanks for your explanation. LGTM.

Reply:
Would this help?
```go
if (mathutil.IntBits == 64 && d == math.MinInt64) || d == math.MinInt32)
```

Reply:
that is great, we can use 
```go
if d == mathutil.MinInt {
	return 0, false, nil
}
```

Reply:
updated